msu/systems/mod_settings/settings_page.nut

```diff
@@ -114,6 +114,14 @@
 		return this.getSettings().values();
 	}
 
+	function resetSettings()
+	{
+		foreach (setting in this.getAllSettingsAsArray())
+		{
+			if (setting instanceof ::MSU.Class.AbstractSetting) setting.reset();
```

Comment:
Why is the `if` necessary here?

Reply:
Because there are also 'settings' that are not children of AbstractSetting due to being visual elements.
Which maybe shouldn't be a thing. @Enduriel

Reply:
Should we add a second function called getAllElementsAsArray and have the two be distinct? Imo this is a good idea

Reply:
Yeah ig that's cleaner

Reply:
Ah but then we have two very similar functions that have to be duplicated everywhere, that's also not so nice